matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/MXOlmDevice.kt

```diff
@@ -421,6 +428,34 @@ internal class MXOlmDevice @Inject constructor(
         return null
     }
 
+    fun storeOutboundGroupSessionForRoom(roomId: String, sessionId: String) {
+        outboundGroupSessionStore[sessionId]?.let {
+            store.storeCurrentOutboundGroupSessionForRoom(roomId, it)
+        }
+    }
+
+    fun restoreOutboundGroupSessionForRoom(roomId: String):  MXOutboundSessionInfo? {
+        val restoredOutboundGroupSession = store.getCurrentOutboundGroupSessionForRoom(roomId)
+        if (restoredOutboundGroupSession != null) {
+            val sessionId = restoredOutboundGroupSession.outboundGroupSession.sessionIdentifier()
+            if (!outboundGroupSessionStore.containsKey(sessionId)) {
+                outboundGroupSessionStore[sessionId] = restoredOutboundGroupSession.outboundGroupSession
+                return MXOutboundSessionInfo(
+                        sessionId = sessionId,
+                        sharedWithHelper = SharedWithHelper(roomId, sessionId, store),
+                        restoredOutboundGroupSession.creationTime
+                )
+            } else {
+                restoredOutboundGroupSession.outboundGroupSession.releaseSession()
+            }
+        }
+        return null
+    }
+
+    fun discardOutboundGroupSessionForRoom(roomId: String) {
+        outboundGroupSessionStore.remove(roomId)?.releaseSession()
```

Comment:
Shouldn't be `outboundGroupSessionStore.remove(sessionId)?.releaseSession()` here as session IDs are the keys?

Reply:
Very good catch! I agree with you

Reply:
Updated